pkg/reconciler/ingress/ingress.go

```diff
@@ -623,3 +664,130 @@ func isRouteReady(r *routev1.Route) bool {
 	}
 	return false
 }
+
+func (r *Reconciler) reconcileSmmr(ctx context.Context, ing *v1alpha1.Ingress) error {
+	logger := logging.FromContext(ctx)
+
+	if err := r.reconcileNetworkPolicy(ctx, ing, false); err != nil {
+		return err
+	}
+
+	// update ServiceMeshMemberRole with the namespace info where knative routes created
+	smmr, err := r.smmrLister.ServiceMeshMemberRolls(smmrNamespace).Get(smmrName)
+	if err != nil {
+		return fmt.Errorf("failed to get SMMR in %q: %w", smmrNamespace, err)
+	}
+	newMembers, changed := appendIfAbsent(smmr.Spec.Members, ing.GetNamespace())
+	if changed {
+		existing := smmr.DeepCopy()
+		existing.Spec.Members = newMembers
+		if _, err := r.smmrClient.MaistraV1().ServiceMeshMemberRolls(existing.Namespace).Update(existing); err != nil {
+			// ref for substring https://github.com/Maistra/istio-operator/blob/maistra-1.0/pkg/controller/servicemesh/validation/memberroll.go#L95
+			if strings.Contains(err.Error(), "one or more members are already defined in another ServiceMeshMemberRoll") {
```

Comment:
I think it'd be super neat to surface this in the status of the Ingress to make it more actionable for the user. A potential seperate PR though imo.

Reply:
Yes, agreed. I opened the ticket for it https://issues.redhat.com/browse/SRVKS-377